Samples.Objc/Samples.ObjC/ViewControllers/CheckoutViewController/CheckoutViewController.m

```diff
@@ -0,0 +1,376 @@
+//
+//  CheckoutViewController.m
+//  PayPalNativeCheckoutObjC
+//
+//  Created by Nguyen, The Nhat Minh on 3/9/21.
+//  Copyright © 2020 PayPal. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "CheckoutViewController.h"
+#import "AddItemViewController.h"
+#import "AddItemCell.h"
+#import "ItemCell.h"
+#import "TotalCell.h"
+#import "FetchAccessTokenEndpoint.h"
+#import "CreateOrderEndpoint.h"
+#import "AccessTokenResponse.h"
+#import "CreateOrderResponse.h"
+#import "PayPalAPI.h"
+@import PayPalCheckout;
+
+@interface CheckoutViewController ()
+@property (nonatomic) UISegmentedControl *segmentedControl;
+@property (nonatomic) UITableView *tableView;
+@property (nonatomic) UIButton *checkoutButton;
+@property (nonatomic) NSMutableArray<PPCPurchaseUnitItem *> *items;
+@end
+
+@implementation CheckoutViewController
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+  [self setupSampleItem];
+  [self setupUI];
+  [self setupTableView];
+  [self setupConstraints];
+  [self setupPayPalCheckoutCallbacks];
+}
+
+- (void)setupPayPalCheckoutCallbacks {
+  [PPCheckout setOnApproveCallback:^(PPCApproval *approval) {
+    if ([approval.data.intent isEqualToString:@"AUTHORIZE"]) {
+      [approval.actions authorizeOnComplete:^(PPCAuthorizeActionSuccess *success, NSError *error) {
+        if (error) {
+          NSLog(@"Fail to authorize order with error %@", error.localizedDescription);
+        } else if (success) {
+          NSLog(@"Authorize order successfully");
+        } else {
+          NSLog(@"Authorize order: No error and no success response");
+        }
+      }];
+    } else if ([approval.data.intent isEqualToString:@"CAPTURE"] || [approval.data.intent isEqualToString:@"SALE"]) {
+      [approval.actions captureOnComplete:^(PPCCaptureActionSuccess *success, NSError *error) {
+        if (error) {
+          NSLog(@"Fail to capture order with error %@", error.localizedDescription);
+        } else if (success) {
+          NSLog(@"Capture order successfully");
+        } else {
+          NSLog(@"Capture order: No error and no success response");
+        }
+      }];
+    }
```

Comment:
Can we maybe add these closures to the `start` function? For clarity sake, i think demonstrating a single closure setting mechanism in a single flow would be best. If we want to demonstrate multiple ways of setting the configs, i say we do that in multiple flows. What do you think?

Reply:
Hmmm, I thought about that, but then I decided to leave them out of `start` mostly for adding `PaymentButton` to the sample app in the future.
Since with `PaymentButton` we won't be calling `start`, and we'll need to set callbacks this way.

Reply:
oohh I see, that makes sense!!!! 
I'll put these closures in `start`!!!!!